cnf-certification-test/platform/suite.go

```diff
@@ -282,18 +282,20 @@ func testTainted(env *provider.TestEnvironment) {
 		tnf.ClaimFilePrintf("Node %s kernel is tainted. Taints mask=%d - Decoded taints: %v",
 			nodeName, taintsMask, nodetainted.DecodeKernelTaintsFromBitMask(taintsMask))
 
-		// Check the white list. If empty, mark this node as failed.
-		if len(whiteListedModules) == 0 {
-			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewNodeReportObject(nodeName, "No modules in whitelist", false))
+		// Check the allow list. If empty, mark this node as failed.
+		if len(allowListedModules) == 0 {
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewNodeReportObject(nodeName, "Node contains taints not covered by module allowlist", false).
```

Comment:
In my opinion, we should have 2 levels of report objects for this test case:
- a node level (top level), this includes internal errors (prefix internal errors with some text like "internal error: "?).
- a node taint level, that would include tainted modules or other taints. In this level, the message should include the taint text and bit in human readable format
We basically need to capture what the following log captured before. I would remove the old log once the info is converted to failure reason.
	logrus.Infof("Nodes with errors: %+v", errNodes)
	logrus.Infof("Bad Modules: %+v", badModules)
	logrus.Infof("Taints not related to any module: %+v", otherTaints)
This is the current output:	
STEP: Checking kernel taints of node osgh1-0 @ 08/21/23 12:16:52.017
  Node osgh1-0 kernel is tainted. Taints mask=512 - Decoded taints: [kernel issued warning (tainted bit 9)]
  Node osgh1-0 - taint bit 9 is set but it's not caused by any module.
  STEP: Checking kernel taints of node osgh1-2 @ 08/21/23 12:16:53.116
  Node osgh1-2 kernel is tainted. Taints mask=512 - Decoded taints: [kernel issued warning (tainted bit 9)]
  Node osgh1-2 - taint bit 9 is set but it's not caused by any module.
  STEP: Checking kernel taints of node osgh1-1 @ 08/21/23 12:16:54.257
  Node osgh1-1 kernel is tainted. Taints mask=512 - Decoded taints: [kernel issued warning (tainted bit 9)]
  Node osgh1-1 - taint bit 9 is set but it's not caused by any module.
INFO   [Aug 21 12:16:55.402][suite.go: 328] Nodes with errors: []                        
INFO   [Aug 21 12:16:55.402][suite.go: 329] Bad Modules: map[]                           
INFO   [Aug 21 12:16:55.402][suite.go: 330] Taints not related to any module: map[osgh1-0:[9] osgh1-1:[9] osgh1-2:[9]]

Reply:
Okay updated!